log4j2/docs/log4j2-tutorial.md

```diff
@@ -0,0 +1,153 @@
+﻿# Log4J 2
+## 1 Características
+### 1.1 Introducción
+* Log4j2 compatible y seguro ya que usa las mismas funcionalidades de slf4j.
+* Log4j2 más extenso que slf4j.
+* Log4j2 escalable porque puede soportar otras implementaciones. 
+* Log4j2 incluye integración mediante el módulo o puente "log4j-to-slf4j" para cambiar a slf4j cuando se requiera.
+* No permite multiples implementaciones.
+* Algunas de las funcionalidades de Log4j2:
+	* Message API
+	* Java 8 lambda support
+	* Mixing {}-style
+	*  CloseableThreadContext 
+	* Log4j2's ThreadContext
+	* SLF4J does not support the FATAL log level.
+	* Log4j2 has support for custom log levels.
+	* The Log4j2 API accepts any Object, not just Strings
+	* The above you get for free just by using the Log4j2 API directly. 
+	* SLF4J Markers'
+## Configuración
+### Ecosistema
+* Existen diversas herramientas para la administración de proyectos, con tipos de instalación tradicional como: Maven, Gradle, Ivy, etc.
+* Para esta versión se tienen diversas `appenders` para poder configurar la bitácora de una manera más clara y consciza, a continucación se listan algunos:
+	1.  CouchDB
+	2. MongoDB
+	3. Cassandra
+	4. IO Streams
+	5. Scala API
+* En la siguiente liga se encuentrar la configuración tradicional y appenders mencionados en el punto anterior: https://logging.apache.org/log4j/2.x/maven-artifacts.html
+### Archivo de configuración o implementación
+* Existen diversos formatos para realizar la configuración de Log4J :
+	1. Json
+	2. Yaml
+	3. Xml
+	4. Archivo de propiedades "fileName.properties".
+	* Para configurar cada una de las implementaciones visitar la liga correspondiente.
+		1. `Json` https://logging.apache.org/log4j/2.x/manual/configuration.html#JSON
+		2. `Yaml`https://logging.apache.org/log4j/2.x/manual/configuration.html#YAML
+		3. `Xml` https://logging.apache.org/log4j/2.x/manual/configuration.html#XML
+				 
+* Log4J2 cuenta con una configuración por default que es aplicada en caso de no encontrar archivos de configuración.
+* Log4J2 define la clase `ConfigurationFactory` encargada de detectar la presencia de archivos de configuración con base a las siguientes reglas:
+	1. Se intenta ubicar el archivo especificado por una JVM system property:  `log4j.configurationFile`
+	2. En caso de no existir el archivo especificado en el punto anterior, la clase `ConfigurationFactory` buscará `log4j2-test.properties` in the classpath.
```

Comment:
¿in the classpath?  ¿spanglish?   evita hacer copy & paste de la documentación. Traducir esta parte. Revisar toda la lista ya que aparece varias veces.